### PR TITLE
Updated gradle dependencies

### LINKED ISSUE #2366: Update library dependencies in the application.
**Steps to reproduce it**

There are many library dependencies for the application that needs to be updated to newer and latest versions. This gives us the latest and more stable code from the libraries and thus makes the application more stable. 

**Would you like to work on the issue?**
Yes

**Target Milestone to solve the issue**
It can be solved in a small fix.

### PR DESCRIPTION
**Pull request by sriramr98**
Fixes #2366
Changes: Updated the versions of outdated dependencies.

### CODE REVIEW

**Comment by sriramr98** [comment]
@iamareebjamal  please review the PR

File: android/app/build.gradle [hunk 1]
```diff
@@ -103,26 +103,26 @@ dependencies {
     implementation 'com.squareup.retrofit2:retrofit:2.3.0'
```
Comment (iamareebjamal) [inline android/app/build.gradle:103]: 'com.github.jasminb:jsonapi-converter:0.8.1'

File: android/app/build.gradle [hunk 2]
```diff
@@ -103,26 +103,26 @@ dependencies {
...
     implementation 'com.squareup.retrofit2:converter-jackson:2.3.0'
     implementation 'com.squareup.retrofit2:adapter-rxjava2:2.3.0'
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.8.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.9.0'
```
Comment (iamareebjamal) [inline android/app/build.gradle]: 3.10.0

File: android/app/build.gradle [hunk 3]
```diff
@@ -103,26 +103,26 @@ dependencies {
...
     implementation 'com.facebook.stetho:stetho:1.5.0'
     implementation 'com.facebook.stetho:stetho-okhttp3:1.5.0'
     implementation 'com.uphyca:stetho_realm:2.1.0'
-    implementation 'io.reactivex.rxjava2:rxandroid:2.0.1'
+    implementation 'io.reactivex.rxjava2:rxandroid:2.0.2'
     implementation 'com.github.yalantis:ucrop:2.2.1'
-    implementation 'io.reactivex.rxjava2:rxjava:2.1.3'
-    implementation "android.arch.lifecycle:reactivestreams:1.0.0"
-    kapt 'com.jakewharton:butterknife-compiler:8.6.0'
+    implementation 'io.reactivex.rxjava2:rxjava:2.1.9'
```
Comment (iamareebjamal) [inline android/app/build.gradle]: 2.1.10

File: android/app/build.gradle [hunk 4]
```diff
@@ -103,26 +103,26 @@ dependencies {
...
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.8.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.9.0'
     implementation 'com.squareup.picasso:picasso:2.5.2'
     implementation 'com.jakewharton.picasso:picasso2-okhttp3-downloader:1.1.0'
     implementation 'com.jakewharton.threetenabp:threetenabp:1.0.5'
     implementation 'com.squareup:otto:1.3.8'
-    implementation 'com.jakewharton.timber:timber:4.5.1'
-    implementation 'com.jakewharton:butterknife:8.6.0'
+    implementation 'com.jakewharton.timber:timber:4.6.0'
+    implementation 'com.jakewharton:butterknife:8.8.1'
```
Comment (iamareebjamal) [inline android/app/build.gradle:112]: Move its kapt version together

File: android/app/build.gradle [hunk 5]
```diff
@@ -103,26 +103,26 @@ dependencies {
...
     implementation 'com.facebook.stetho:stetho-okhttp3:1.5.0'
     implementation 'com.uphyca:stetho_realm:2.1.0'
-    implementation 'io.reactivex.rxjava2:rxandroid:2.0.1'
+    implementation 'io.reactivex.rxjava2:rxandroid:2.0.2'
     implementation 'com.github.yalantis:ucrop:2.2.1'
-    implementation 'io.reactivex.rxjava2:rxjava:2.1.3'
-    implementation "android.arch.lifecycle:reactivestreams:1.0.0"
-    kapt 'com.jakewharton:butterknife-compiler:8.6.0'
+    implementation 'io.reactivex.rxjava2:rxjava:2.1.9'
+    implementation "android.arch.lifecycle:reactivestreams:1.1.0"
```
Comment (iamareebjamal) [inline android/app/build.gradle]: Move arch components together

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
'org.mockito:mockito-core:2.15.0'

**Comment by iamareebjamal** [comment]
Please don't close PRs like this. This is against the best practices at FOSSASIA

**Comment by sriramr98** [comment]
Sorry. There was a problem in the branch and I had to delete it. Will be more careful in the future. Will also send a new PR for the issue.

**Comment by iamareebjamal** [comment]
git allows you to robustly handle commits in branches with rebasing. You don't need to delete a branch to fix issues in it

**Comment by sriramr98** [comment]
Yeah but I am still learning git and I didn't wanna bloat the commit history.

**Comment by iamareebjamal** [comment]
Commit history doesn't get bloated in rebasing

**Pull request closed without merging**